### PR TITLE
Fix the handling of shared library dependencies in subfolders.

### PR DESCRIPTION
**Pull request by gl84**
Atm the linking fails with a hard error for dependencies of dynamic libraries that are in subdirectories.

### CODE REVIEW

**Review by kripken — APPROVED** [review]
lgtm

@sbc100 , any concerns?

File: emcc.py [hunk 1]
```diff
@@ -670,9 +670,10 @@ def process_dynamic_libs(dylibs, lib_dirs):
       path = find_library(needed, lib_dirs)
       if path:
         extras.append(path)
+        seen.add(needed)
       else:
-        exit_with_error(f'{dylib}: shared library dependency not found: `{needed}`')
-      to_process.append(needed)
+        exit_with_error(f'{os.path.basename(dylib)}: shared library dependency not found: `{needed}`')
```
Comment (sbc100) [inline emcc.py:675]: Why this change?   Isn't useful to see the full filename?

File: tests/test_other.py [hunk 2]
```diff
@@ -7256,6 +7256,13 @@ def test_side_module_transitive_deps(self):
     err = self.expect_fail(final_link)
     self.assertContained('error: libside2.wasm: shared library dependency not found: `libside1.wasm`', err)
 
+  def test_side_module_folder_deps(self):
+    # Build side modules in a subfolder
+    os.mkdir('subdir')
+    self.run_process([EMCC, test_file('hello_world.c'), '-s', 'SIDE_MODULE', '-o', 'subdir/libside1.so'])
+    self.run_process([EMCC, test_file('hello_world.c'), '-s', 'SIDE_MODULE', '-o', 'subdir/libside2.so', '-L', 'subdir', '-lside1'])
+    self.run_process([EMCC, test_file('hello_world.c'), '-s', 'MAIN_MODULE', '-o', 'subdir/main.js', '-L', 'subdir', '-lside2'])
```
Comment (sbc100) [inline tests/test_other.py:7264]: Nice!   

Maybe `-o main.js` to that the final output is not itself in the subdir, just to make sure that works too?

**Review by sbc100 — APPROVED** [review]

File: tests/test_other.py (hunk 2 above)
Comment (gl84) [reply tests/test_other.py:7264]: Sounds good, changed to -o main.js

File: emcc.py [hunk 3]
```diff
@@ -670,9 +670,10 @@ def process_dynamic_libs(dylibs, lib_dirs):
       path = find_library(needed, lib_dirs)
       if path:
         extras.append(path)
+        seen.add(needed)
       else:
-        exit_with_error(f'{dylib}: shared library dependency not found: `{needed}`')
-      to_process.append(needed)
+        exit_with_error(f'{os.path.normpath(dylib)}: shared library dependency not found: `{needed}`')
```
Comment (sbc100) [inline emcc.py:675]: I'm not sure why normpath is useful here either.. but I guess its no harm.

**Review by sbc100 — APPROVED** [review]

**Comment by kripken** [comment]
I restarted the test here, and it failed again on `sockets.test_enet`. I can't be sure this PR doesn't affect linking, so please merge in latest `main` to here, as it may fix the error if it was a previous failure on trunk.

**Comment by gl84** [comment]
Rebased on current tot.

Commit affects only linking, no idea why wasm2js1 timed out...
